tests/api/test_authenticator.py

```diff
@@ -3026,3 +3028,64 @@ def test_oauth_authentication_invalid_token(self):
             self._db, "Bearer - this is a bad token"
         )
         assert INVALID_OAUTH_BEARER_TOKEN == problem
+
+
+class TestSirsiDynixAuthenticationProvider(AuthenticatorTest):
+    def setup_method(self):
+        super().setup_method()
+        self.integration = self._external_integration(
+            "api.sirsidynix",
+            goal=ExternalIntegration.PATRON_AUTH_GOAL,
+            settings={
+                ExternalIntegration.URL: "http://example.org/sirsi",
+                SirsiDynixAuthenticationProvider.Keys.APP_ID: "appid",
+                SirsiDynixAuthenticationProvider.Keys.CLIENT_ID: "clientid",
+            },
+        )
+        self.api = SirsiDynixAuthenticationProvider(
+            self._default_library, self.integration
+        )
+
+    def _headers(self):
+        return {
+            "SD-Originating-App-Id": self.api.sirsi_app_id,
+            "x-sirs-clientID": self.api.sirsi_client_id,
+        }
+
+    def test_settings(self):
+        # with leading slash
+        assert self.api.server_url == "http://example.org/sirsi/"
+        assert self.api.sirsi_client_id == "clientid"
+        assert self.api.sirsi_app_id == "appid"
+
+    @patch("api.sirsidynix_authentication_provider.HTTP.request_with_timeout")
+    def test_api_patron_login(self, mock_request):
+        response_dict = {"sessionToken": "xxxx"}
+        mock_request.return_value = MockRequestsResponse(200, content=response_dict)
+
+        response = self.api.api_patron_login("username", "pwd")
+
+        assert mock_request.call_count == 1
+        assert mock_request.call_args_list[0] == call(
```

Comment:
Could we use `call_args` -- rather than `call_args_list[0]` -- here, since we're only expecting one call?

```assert mock_request.call_args == call(...)```

Reply:
That works too, updated.